edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/routing/RoutingEdgeHub.cs

```diff
@@ -57,9 +57,13 @@ public Task ProcessDeviceMessage(IIdentity identity, IMessage message)
 
         public Task ProcessDeviceMessageBatch(IIdentity identity, IEnumerable<IMessage> messages)
         {
-            IEnumerable<IRoutingMessage> routingMessages = Preconditions.CheckNotNull(messages)
-                .Select(m => this.ProcessMessageInternal(m, true));
-            return this.router.RouteAsync(routingMessages);
+            Metrics.MessageCount(identity, messages.Count());
+            using (Metrics.MessageLatency(identity))
```

Comment:
Does this latency take into account multiple messages?

Reply:
I'm removing this for now. I'll think about how to handle it and send out a separate PR for that. Good catch.